src/modules/hip/kernel/to_decibels.hpp

```diff
@@ -0,0 +1,313 @@
+                                             float *inverseMagnitudeTensor)
+
+{
+    int id_x = hipBlockIdx_x * hipBlockDim_x + hipThreadIdx_x;
+    int id_z = hipBlockIdx_z * hipBlockDim_z + hipThreadIdx_z;
+
+    // Do final reduction on block wise max
+    if (computeMax)
+    {
+        uint srcIdx = id_z * maxLength;
+        __shared__ float max_smem[256];
+        max_smem[hipThreadIdx_x] = srcPtr[srcIdx];
+
+        if (id_x >= maxLength)
+            return;
+
+        srcIdx += id_x;
+        float maxVal = srcPtr[srcIdx];
+        while (id_x < maxLength)
+        {
+            maxVal = fmaxf(maxVal, srcPtr[srcIdx]);
+            id_x += hipBlockDim_x;
+            srcIdx += hipBlockDim_x;
+        }
+        max_smem[hipThreadIdx_x] = maxVal;
+        __syncthreads();
+
+        // do reduction on min_smem and max_smem
+        for (int threadMax = 128; threadMax >= 1; threadMax /= 2)
+        {
+            if (hipThreadIdx_x < threadMax)
+                max_smem[hipThreadIdx_x] = max(max_smem[hipThreadIdx_x], max_smem[hipThreadIdx_x + threadMax]);
+            __syncthreads();
+        }
+
+        if (hipThreadIdx_x == 0)
+            inverseMagnitudeTensor[id_z] = 1.f / max_smem[0];
+    }
+    else
+    {
+        inverseMagnitudeTensor[id_z] = 1.0f;
+    }
+}
+
+__global__ void max_reduction_hip_tensor(float *srcPtr,
+                                         uint2 srcStridesNH,
+                                         RpptImagePatchPtr srcDims,
+                                         uint numDims,
+                                         float *maxArr)
+{
+    int id_y = hipBlockIdx_y * hipBlockDim_y + hipThreadIdx_y;
+    int id_z = hipBlockIdx_z * hipBlockDim_z + hipThreadIdx_z;
+
+    // for 1D input
+    if (numDims == 1)
+    {
+        int id_x = (hipBlockIdx_x * hipBlockDim_x + hipThreadIdx_x) * 8;
+        uint srcLength = srcDims[id_z].height;
+        uint srcIdx = id_z * srcStridesNH.x;
+        __shared__ float max_smem[512];
+        max_smem[hipThreadIdx_x] = srcPtr[srcIdx];
+
+        if (id_x >= srcLength)
+            return;
+
+        srcIdx += id_x;
+        d_float8 src_f8;
+        rpp_hip_load8_and_unpack_to_float8(srcPtr + srcIdx, &src_f8);
+        rpp_hip_math_max8(&src_f8, &max_smem[hipThreadIdx_x]);
+        __syncthreads();
+
+        // do reduction on min_smem and max_smem
+        for (int threadMax = 256; threadMax >= 1; threadMax /= 2)
+        {
+            if (hipThreadIdx_x < threadMax)
+                max_smem[hipThreadIdx_x] = fmaxf(max_smem[hipThreadIdx_x], max_smem[hipThreadIdx_x + threadMax]);
+            __syncthreads();
+        }
+
+        if (hipThreadIdx_x == 0)
+        {
+            int dstIdx = id_z * hipGridDim_x + hipBlockIdx_x;
+            maxArr[dstIdx] = max_smem[0];
+        }
+    }
+    // for 2D input
+    else if (numDims == 2)
+    {
+        int id_x = hipBlockIdx_x * hipBlockDim_x + hipThreadIdx_x;
```

Comment:
As discussed in call
2D case is not vectorized due to some constraints from rocAL on not being able to allocate additional aligned memory
in case of 2D